src/platform/stub-setup.ts

```diff
@@ -16,6 +19,23 @@ const smarthomeStub: {
   },
 };
 
-export function injectSmarthomeStubs() {
+// Injects stubs into the global scope, allowing implicit usage for testing
+export function injectSmarthomeStubs(): void {
   (global as any).smarthome = smarthomeStub;
-}
\ No newline at end of file
+}
+
+// Module-level variable to hold the active instance the AppStub
+var activeAppStub: AppStub | undefined;
+
+// Module-level getter for the active AppStub
+export function getActiveAppStub(): AppStub {
```

Comment:
Sorry for lack of details with my earlier comments, when I mentioned `top-level module function`, I meant function that would take a `smarthome.App` as parameter and promote it to a `smarthome.AppStub` to get the platform out of it and dispatch an intent.

I don't believe you need to get keep any global state, as the test already know about the app they need to test.

Reply:
Oh I see.  Done. The only thing is now the developer will need to re-implement the creation of their App in the same context as their test (as opposed to being able to use the App creation code in their entry point).  However, like you said I think it's something that can be layered later.

Reply:
Wouldn't the app creation automatically use the stub as long as install the `smarthome` namespace before loading the app?